### ISSUE TITLE
Security problem with `request.access_route`

### ISSUE THREAD

**Issue by QuantumGhost** [alisaifee/flask-limiter#41]
According to the documentation of werkzeug, `request.access_route` is computed by looking at forwarded header in the request. However, as malicious users  can submit request with `X-Forwarded-For` of any value, they can bypass limiters easily.

I think that this security problem should be documented clearly, and `utils.get_ipaddr` should be used with caution.

Here is an article discussing this problem: http://esd.io/blog/flask-apps-heroku-real-ip-spoofing.html.

**Comment by alisaifee** [comment]
Thanks, you're right. Do you think using `request.remote_addr` as the value of the request's ip address would be a safer approach. I'm thinking of issuing a warning for anyone using the `utils.get_ipaddr` method and deprecating it over time. The documentation will then point to a recipe of how to get the ip address of the request with/without a proxy.

**Comment by tonymillion** [comment]
:+1: for this - I've just dealt with the same issue

Even though we used the werkzeug `ProxyFix` middleware, the default ratelimiter `get_ipaddr` was bypassing that and using a spoofed address (`request.access_route[0]`).

I'd suggest putting a big warning about using `ProxyFix` if you're behind a loadbalancer (in our case AWS ELB) and change the `get_ipaddr` to use `request.remote_add`

**Comment by alisaifee** [comment]
@tonymillion & @QuantumGhost thank you for the feedback.

I've finally added the deprecation and some documentation about the problem [here](http://flask-limiter.readthedocs.org/en/latest/#rate-limit-domain). The code remains backward compatible for now, but I will eventually remove the default key function behaviour.

Do let me know if the documentation I've added can be improved.

**Comment by marians** [comment]
The change to ProxyFix, as explained by http://esd.io/blog/flask-apps-heroku-real-ip-spoofing.html, has landed in Werkzeug 0.11.10 and is available to users of the most recent Flask version. I suggest to extend the docs to show and explain the example

```python
from werkzeug.contrib.fixers import ProxyFix
...
app.wsgi_app = ProxyFix(app.wsgi_app, num_proxies=1)
```

That increases the likelihood that developers of applications behind proxies pick up this important piece of information.
